pkg/build/generator/generator.go

```diff
@@ -670,6 +671,48 @@ func (g *BuildGenerator) resolveImageSecret(ctx kapi.Context, secrets []kapi.Sec
 	return nil
 }
 
+//getBuildServiceAccount returns serviceaccount used by new build
+func (g *BuildGenerator) getServiceAccount(buildConfig *buildapi.BuildConfig) string {
+	serviceAccount := buildConfig.Spec.ServiceAccount
+	if len(serviceAccount) == 0 {
+		serviceAccount = g.DefaultServiceAccountName
+	}
+	if len(serviceAccount) == 0 {
+		serviceAccount = bootstrappolicy.BuilderServiceAccountName
+	}
+	return serviceAccount
+}
```

Comment:
the default service account to use can be passed in as an argument to getServiceAccount.